pkg/geo/encode.go

```diff
@@ -311,3 +312,57 @@ func StringToByteOrder(s string) binary.ByteOrder {
 		return DefaultEWKBEncodingFormat
 	}
 }
+
+// LineFromEncodedPolyline takes the encoded polyline ASCII and precision, decodes the points and returns them as a geometry
+func LineFromEncodedPolyline(encodedPolyline string, precision int) (Geometry, error) {
+	flatCoords := decodePolylinePoints(encodedPolyline, precision)
+	ls := geom.NewLineStringFlat(geom.XY, flatCoords).SetSRID(4326)
+
+	g, err := MakeGeometryFromGeomT(ls)
+	if err != nil {
+		return Geometry{}, fmt.Errorf("parsing geography error: %v", err)
+	}
+	return g, nil
+}
+
+func decodePolylinePoints(encoded string, precision int) []float64 {
```

Comment:
Can you add a comment to the function which describes how the algorithm works?

Reply:
Done

Reply:
But wait, the link to the algorithm is already added in the functions under `geo_builtins`. Do you want me to add it here anyway?